superset/assets/cypress/integration/dashboard/dashboard_controls.js

```diff
@@ -0,0 +1,52 @@
+import { WORLD_HEALTH_DASHBOARD, getDashboardUnfavorEndpoing } from './dashboard.helper';
+
+describe('dashboard top-level controls', () => {
+  let sliceIds = [];
+  let dashboard = {};
+
+  beforeEach(() => {
+    cy.server();
+    cy.login();
+
+    cy.visit(WORLD_HEALTH_DASHBOARD);
+
+    cy.get('#app').then((data) => {
+      const bootstrapData = JSON.parse(data[0].dataset.bootstrap);
+      dashboard = bootstrapData.dashboard_data;
+      sliceIds = dashboard.slices.map(slice => (slice.slice_id));
+    });
+  });
+
+  it('should allow favor/unfavor', () => {
+    // make sure dashboard is not favorite
+    cy.request(getDashboardUnfavorEndpoing(dashboard.id));
+    cy.visit(WORLD_HEALTH_DASHBOARD);
+
+    cy.get('.favstar').find('i').should('have.class', 'fa-star-o');
+    cy.get('.favstar').trigger('click');
+    cy.get('.favstar').find('i').should('have.class', 'fa-star')
+      .and('not.have.class', 'fa-star-o');
+
+    cy.get('.favstar').trigger('click');
+    cy.get('.fave-unfave-icon').find('i').should('have.class', 'fa-star-o')
+      .and('not.have.class', 'fa-star');
+  });
+
+  it('should allow auto refresh', () => {
+    const aliases = [];
+    sliceIds
+      .forEach((id) => {
+        const alias = `getJson_${id}`;
+        aliases.push(`@${alias}`);
+
+        cy.route('POST', `/superset/explore_json/?form_data={"slice_id":${id}}`, {}).as(alias);
+      });
+
+    cy.get('#save-dash-split-button').trigger('click');
+    cy.contains('Force refresh dashboard').click();
+
+    cy.wait(aliases).then((xhrs) => {
+      expect(xhrs.length).eq(sliceIds.length);
```

Comment:
Is there an easy way to verify that these slices were not loaded from the cache?

Reply:
true, i should check force refresh parameter in each request.

Reply:
I added check for `force=true` parameter in the request, and `is_cached: false` in the response.